tsdb/tsm1/engine_measurement_schema.go

```diff
@@ -52,18 +68,30 @@ func (e *Engine) MeasurementNames(ctx context.Context, orgID, bucketID influxdb.
 				}
 
 				key, _ := SeriesAndFieldFromCompositeKey(sfkey)
-				name, err := models.ParseMeasurement(key)
+				nameBytes, err := models.ParseMeasurement(key)
 				if err != nil {
 					e.logger.Error("Invalid series key in TSM index", zap.Error(err), zap.Binary("key", key))
 					continue
 				}
 
-				if _, ok := tsmValues[string(name)]; ok {
-					continue
+				name := string(nameBytes)
```

Comment:
I would recommend you always use this in performance sensitive code (such as this loop):

```go
if _, ok := names[string(nameBytes)] ...
```

As Go won't allocate the string here. Only when you add it to the map, will it allocate a string:

```go
names[string(name)] = struct{}{}
```